lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -668,103 +781,68 @@ createSchedule(scf::ForOp forOp, int numStages,
     return operands;
   };
 
-  // Find dependencies with distance of 1.
-  SmallVector<DenseSet<Operation *>> distanceOneUsers(numStages);
-  for (int stage = 0; stage < numStages - 1; stage++) {
-    auto &group = insertAndDeps[stage];
-    for (Operation *op : group) {
-      for (Value operand : getNestedOperands(op)) {
-        if (auto arg = operand.dyn_cast<BlockArgument>()) {
-          if (arg.getArgNumber() > 0 && arg.getOwner() == op->getBlock()) {
-            auto yieldOp = op->getBlock()->getTerminator();
-            Value v = yieldOp->getOperand(arg.getArgNumber() - 1);
-            Operation *defOp = v.getDefiningOp();
-            if (defOp && group.count(defOp) == 0) {
-              // Schedule distance 1 users in the next stage.
-              distanceOneUsers[stage].insert(defOp);
+  // Find dependencies with distance of 1. They will go to the next stage,
```

Comment:
should that also be done in the coarse schedule?

Reply:
Hmm, good point!

Reply:
Actually that would mean that we need to add the regular dependencies also on the coarse level. This kind of defeats the purpose of the coarse schedule - it was supposed to have only the "anchor ops" that we really care about. Their dependencies (and I think the dist 1 deps) are discovered as part of building the final schedule. What do you think?